src/shared/lib/resources/proposal/sprint-with-us.ts

```diff
@@ -0,0 +1,194 @@
+  questionsScore?: number;
+  challengeScore?: number;
+  scenarioScore?: number;
+  priceScore?: number;
+}
+
+export interface SWUProposalPhase {
+  members: SWUProposalTeamMember[];
+  proposedCost: number;
+}
+
+export interface SWUProposalTeamMember {
+  member: UserSlim;
+  scrumMaster: boolean;
+  pending: boolean;
+}
+
+export interface SWUProposalReference {
+  name: string;
+  company: string;
+  phone: string;
+  email: string;
+  order: number;
+}
+
+export interface SWUProposalTeamQuestionResponse {
+  response: string;
+  order: number;
+}
+
+export interface SWUProposalHistoryRecord {
+  id: Id;
+  createdAt: Date;
+  createdBy: UserSlim | null;
+  type: ADT<'status', SWUProposalStatus> | ADT<'event', SWUProposalEvent>;
+  note: string;
+}
+
+export type SWUProposalSlim = Omit<SWUProposal, 'history' | 'opportunity' | 'attachments'>;
+
+// Create.
+
+export type CreateSWUProposalStatus
+  = SWUProposalStatus.Draft
+  | SWUProposalStatus.Submitted;
+
+export interface CreateSWUProposalTeamMemberBody {
+  member: Id;
+  scrumMaster: boolean;
+  pending: boolean;
+}
+
+export interface CreateSWUProposalPhaseBody {
+  members: CreateSWUProposalTeamMemberBody[];
+  proposedCost: number;
+}
+
+export type CreateSWUProposalReferenceBody = SWUProposalReference;
+
+export type CreateSWUProposalTeamQuestionResponseBody = SWUProposalTeamQuestionResponse;
+
+export interface CreateRequestBody {
+  opportunity: Id;
+  inceptionPhase?: CreateSWUProposalPhaseBody;
+  prototypePhase?: CreateSWUProposalPhaseBody;
+  implementationPhase: CreateSWUProposalPhaseBody;
+  references: CreateSWUProposalReferenceBody[];
+  attachments: Id[];
+  teamQuestionResponses: CreateSWUProposalTeamQuestionResponseBody[];
+  status: CreateSWUProposalStatus;
+}
+
+export interface CreateSWUProposalPhaseValidationErrors {
+  members?: CreateSWUProposalTeamMemberValidationErrors[];
+  proposedCost?: string[];
+}
+
+export type CreateSWUProposalTeamMemberValidationErrors = ErrorTypeFrom<CreateSWUProposalTeamMemberBody>;
+
+export interface CreateSWUProposalReferenceValidationErrors extends ErrorTypeFrom<CreateSWUProposalReferenceBody> {
+  parseFailure?: string[];
+}
+
+export interface CreateSWUProposalTeamQuestionResponseValidationErrors extends ErrorTypeFrom<CreateSWUProposalTeamQuestionResponseBody> {
+  parseFailure?: string[];
+}
+
+export interface CreateValidationErrors extends BodyWithErrors {
+  attachments?: string[][];
+  inceptionPhase?: CreateSWUProposalPhaseValidationErrors;
+  prototypePhase?: CreateSWUProposalPhaseValidationErrors;
+  implementationPhase?: CreateSWUProposalPhaseValidationErrors;
+  references?: CreateSWUProposalReferenceValidationErrors[];
+  teamQuestionResponses?: CreateSWUProposalTeamQuestionResponseValidationErrors[];
+  opportunity?: string[];
+  status?: string[];
+}
+
+// Update.
+
+export type UpdateRequestBody
+  = ADT<'edit', UpdateEditRequestBody>
+  | ADT<'submit', string>
+  | ADT<'score', string>
```

Comment:
Need to support scoring questions, code challenge, and team scenario

Reply:
Not sure when pricing should be scored by the system? Once the proposal deadline lapses, after questions are scored, or after the scenario is scored? Are disqualified/screened out proposals included in price scoring?

Reply:
I think it will have to be after the code challenge/scenario is scored.  Disqualified/screened out proposals are not included in the price scoring (at least in the current devex).

I'll add ADTs for scoring questions, code challenge, and scenario, but let me know if you think we should modify this further.

Reply:
That works for me. Sounds good re: automatically scoring price once all qualified team scenarios have been scored.